### PR TITLE
[#429] the Prometheus backend integrates with EKG

### PR DESCRIPTION
**Pull request by CodiePP**
description
-----------

- [x] the prometheus backend integrates with the EKG backend

if the configuration contains a valid port (> 0) for `getPrometheusPort`,
then the prometheus adaptor is started.

`cabal new-run example-complex` will provide metrics on http://localhost:12800/metrics

checklist
---------

- [x] compiles (`cabal new-clean; cabal new-build`)
- [x] tests run successfully (`cabal new-test`)
- [ ] documentation added and created (`cd docs; nix-shell --run make`)
- [ ] link to an issue
- [ ] link to an epic
- [ ] add estimate points
- [x] add milestone (the same as the linked issue)

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -32,4 +31,4 @@ source-repository-package
   location: https://github.com/CodiePP/ekg-prometheus-adapter
   tag: 1a258b6df7d9807d4c4ff3e99722223d31a2c320
 
-reorder-goals: True
+reorder-goals: False
```
Comment (dcoutts) [inline cabal.project]: Looks unrelated. Keep in a separate PR if it's needed.

Comment (CodiePP) [reply cabal.project]: left unchanged.
